### ISSUE TITLE
Out-of-date "The server returns the client payload as-is." in interop UnaryCall specification

### ISSUE THREAD

**Issue by nathanielmanistaatgoogle** [grpc/grpc#527]
In test/cpp/interop/test.proto (and src/csharp/GrpcApi/proto/test.proto and src/node/interop/test.proto, see issue #526), the specification of the UnaryCall method is described as "The server returns the client payload as-is.", but when I implemented my server this way, it failed. Studying the code of the other interop servers, the expected behavior appears to be "the server returns a response with a payload of type COMPRESSABLE and of body of null bytes of the length specified by the request's request_size".

Is this right? It appears to work for me, but what's the actual specification? Do the bytes in the body really have to be null? Please (after designating one copy of the file as authoritative, see issue #526 again) update the description with a formal description of the required server behavior.

**Comment by ejona86** [comment]
The comment you describe in test.proto isn't present in test/proto/test.proto. In general, doc/interop-test-descriptions.md should be the source of truth. However, I'm also aware that changes to the document have not been receiving the review and agreement that they need; we need more process for changes to it. Generally only "important" tests and server features required for those tests are likely to be useful. For all the changes to the document that I have reviewed, I have pushed very hard to prevent any need for guessing.

In your particular case, look at the "UnaryCall" feature for the server: https://github.com/grpc/grpc/blob/master/doc/interop-test-descriptions.md#unarycall

**Comment by nathanielmanistaatgoogle** [comment]
I agree both with your points in the large and with the observation that the particular out-of-date specification called out in this issue has long since been resolved.

**Comment by ejona86** [comment]
The text for UnaryCall and the related server-side features has remained effectively unchanged since October 1st, 2014 (albeit, in the internal-only document at that time).

**Comment by nathanielmanistaatgoogle** [comment]
True, but it wasn't authoritative (or even discoverable) at the time this issue was filed; it now is. :-) At the time this issue was filed the comments in the .proto were the most authoritative-appearing documentation of the test and did not match the implementation.

**Comment by ejona86** [comment]
We'll need to agree to disagree. The document was always authoritative and it was one of 3 files in the team's interop testing folder, which was the team-designated place for such items. I had also presented about its existence at multiple of our team meetings. Before that document, there was no "large_unary" or any other interop test. I also consistently got links to it added to other docs that were made that mentioned the test cases (e.g., the test matrix checklist had a link since October, and it that document was being shown every week in a meeting).

I did everything in my power and believe no corners were cut, but unfortunately you didn't get the information. I fully agree that it was unfortunate, but took offense to your view of the situation exactly because of how well exposed and well defined it was. I chalk it up to "it happens."
